build/build-docker-deploy.sh

```diff
@@ -11,7 +11,7 @@ set -euo pipefail
 # itself inside of docker passing "docker" as the argument causing the
 # commands in the if-branch to be executed within the docker
 # container.
-if [ "${1:-}" = "docker" ]; then
+if [ "${1-}" = "docker" ]; then
```

Comment:
`${1-}` is not documented (in `man bash`) as far as I can find (i had to google for this and it was tricky to find), but `${X-Y}` and `${X:-Y}` differ in their handling of empty values for `$X`: both cases resolve to `$X` if it has a non-empty value and `Y` if it is unset. The difference is that when `$X` is explicitly set to an empty string, `${X:-Y}` returns `Y` and `${X-Y}` returns an empty string. It makes no difference when `Y` is also an empty string, but when it does matter I think I'd generally want and expect the `-` behavior instead of the `:=` behavior.